cfme/storage/volume.py

```diff
@@ -251,21 +267,35 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for Volume to disappear, continuing')
 
-    def update(self, updates):
+    def update(self, updates, storage_manager=None, from_manager=None):
```

Comment:
I think overall we need to better use collection/entity filters in order to apply navigation control.  We can discuss this more if you would like.

Here specifically though, I think its a bad pattern to have these two kwargs.  There's no check against `storage_manager` being empty when `from_manager` is `True`, much less a docblock explaining that one needs to be passed when the other, a boolean, is `True`

I think that the storage manager should not be passed into these methods at all, and should be referenced through the relevant object. I'm not sure off-hand if its available through `Volume.provider`, if not, storage_manager should be a Volume attribute passed on instantiation.

In `update` you may have to reference a storage_manager from `updates` itself, in the other crud methods you should be able to use `self.provider.storage_manager` or `self.provider`.\

The other consideration, is that the choice of whether or not to go `from_manager` could be made on whether or not storage_manager is set on a `Volume` instance.  That way there's no kwarg to pass at all, and  navigation could look like:
```
# nav via volume collection
volume = app.collections.storage_volumes.instantiate('test', test_provider)
volume.update({'name': 'updated_test'})

# nav via storage manager
volume = app.collections.storage_volumes.instantiate('test', test_provider, test_storage_manager)
volume.update({'name': 'updated_from_storage_manager'}
```

Reply:
@mshriver I reworked it the way storage_manager doesn't have to be set anywhere: https://github.com/ManageIQ/integration_tests/pull/8457/commits/a90987ca04b7016e53fa6aad628cf5c2aa21fc83
I can improve navigation from_manager/from_volume in next step as I really need to have this merged( saves a lot of work on testing) and then I can start improving it. I already got a list of possible improvements.